Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -406,18 +405,50 @@ function New-LDAPSIdentitySource {
     else {
         Write-Host "No existing external identity sources found."
     }
-
     $Password = $Credential.GetNetworkCredential().Password
-    $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
-    [System.Array]$Certificates = 
-        foreach($CertFile in $DestinationFileArray) {
+    $DestinationFileArray = @()
+    if ($PSBoundParameters.ContainsKey('SSLCertificatesSasUrl')) {
+        $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
+    } else {
+        $exportFolder = "$home/"
+        $remoteComputers = ,$PrimaryUrl
+        if ($PSBoundParameters.ContainsKey('SecondaryUrl')) { $remoteComputers += $SecondaryUrl }
+        
+        foreach ($computerUrl in $remoteComputers) {
+            if ($computerUrl.IndexOf('ldaps://') -ne 0) { Write-Error "Incorrect Url format entered: " + $computerUrl -ErrorAction Stop}
+
+            $TrimedUrl = $computerUrl.substring('ldaps://'.Length)
```

Comment:
This needs a better validation. Use `[Uri]::TryParse` instead and access host FQDN from there.

Reply:
also typo -> It's `Trimmed`

Reply:
Also check that it's not an IP address.

Reply:
could that be possible the url is not ip address but a domain name?

Reply:
No, the certificate will be issued to a FQDN, when the customer tries to use IP the (S)ecure layer of LDAPS rejects the connection because the client side cannot confirm the authenticity of the server. This is a common point of confusion for the customer and we should give them clear error in this case.